### ISSUE TITLE
Error while installing dependencies on macOS

### ISSUE THREAD

**Issue by aleixjf** [Marekkon5/onetagger#41]
Good morning!

I'm trying to compile a version with some changes I've made myself (to try/test adding Spotify too + iTunes Search API (public, limited to 20 queries/minute aprox)), but I'm not being successful to compile it, not even the base version with no modifications/add-ons.

I've already installed rustup, and I had node previously installed. However, when trying to install the dependencies, I get the following message:
**The operation couldn’t be completed. Unable to locate a Java Runtime that supports apt.
Please visit http://www.java.com for information on installing Java.**

I've already installed latest Java's version as per the instructions but no success at all, still same message. Any ideas?
I'm currently using a Mac Mini M1 with macOS Big Sur 11.5.2.

Of course, trying to build the mac version without dependencies returns an error.

**Comment by bascurtiz** [comment]
Cool!

I confirm, just tried on Big Sur 11.4.
Installed `jre-8u301-macosx-x64` from `https://www.java.com/en/download/manual.jsp`  after, but nope.
When simply running `cargo run` ends up in mutliple : `error: linker `x86_64-apple-darwin14-clang` not found`

I informed @Marekkon5 whose back to school since today, so we'll need to be patient.

**Comment by bascurtiz** [comment]
Seems it's totally unrelated to Java actually (and a coincidence Java pops up using `apt` command):
https://stackoverflow.com/questions/66267454/jdk-is-installed-on-mac-but-im-getting-the-operation-couldn-t-be-completed-un

**Comment by aleixjf** [comment]
> Seems it's totally unrelated to Java actually (and a coincidence Java pops up using `apt` command):
> https://stackoverflow.com/questions/66267454/jdk-is-installed-on-mac-but-im-getting-the-operation-couldn-t-be-completed-un

Yeah, I already saw that thread while looking for possible solutions myself, but wasn't successful at finding one... That's why I was asking here. Maybe those dependencies can be directly installed using brew for macOS/linux rather than apt?

**Comment by bascurtiz** [comment]
I tried that replacing `sudo apt` with `brew install` but nope. Too many dependencies fail to be found this way.
Looks like we need Xcode ( https://developer.apple.com/download/all/?q=xcode ) but it's 11gb, and i'm on a limited connection, so I rather wait for Marekkon5 to confirm.

**Comment by aleixjf** [comment]
> I tried that replacing `sudo apt` with `brew install` but nope. Too many dependencies fail to be found this way.
> Looks like we need Xcode ( https://developer.apple.com/download/all/?q=xcode ) but it's 11gb, and i'm on a limited connection, so I rather wait for Marekkon5 to confirm.

I already have XCode + XCode Command Line tools installed, but still I get the same results as you...

**Comment by Marekkon5** [comment]
One Tagger doesn't use java, the `apt` commands are used on Debian based Linux distributions.
However I don't have a Mac and neither of us were able to compile 1T natively for Mac, so there are no instructions/support for this. I recommend trying the instructions from readme in a Ubuntu virtual machine (or PC). Or if you push your changes to GitHub, GitHub actions should automatically build it for you.

**Comment by bascurtiz** [comment]
@aleixjf 
FWIW, I made a video how to compile One Tagger from scratch on macOS (using VirtualBox/Ubuntu):
https://youtu.be/gJO19nN69Y8
